### PR TITLE
VDS-1200, VDS-714

### PR DESCRIPTION
**Pull request by asvishnyakov**

### CODE REVIEW

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+using System.Linq;
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Please rename file itself to ImportProductPricesExistenceValidator.cs

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using FluentValidation.Validators;
+using VirtoCommerce.PricingModule.Core.Model.Search;
+using VirtoCommerce.PricingModule.Core.Services;
+using VirtoCommerce.SimpleExportImportModule.Core.Models;
+
+namespace VirtoCommerce.SimpleExportImportModule.Data.Validation
+{
+    public class ImportProductPricesExistenceValidator: AbstractValidator<ImportProductPrice[]>
+    {
+        public const string ExistingPrices = nameof(ExistingPrices);
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Is this constant must be public?

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+using System.Threading.Tasks;
+using FluentValidation;
+using FluentValidation.Validators;
+using VirtoCommerce.PricingModule.Core.Model.Search;
+using VirtoCommerce.PricingModule.Core.Services;
+using VirtoCommerce.SimpleExportImportModule.Core.Models;
+
+namespace VirtoCommerce.SimpleExportImportModule.Data.Validation
+{
+    public class ImportProductPricesExistenceValidator: AbstractValidator<ImportProductPrice[]>
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Add space before the colon.
P.S. And please enable autoformatter ;)

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        public const string ExistingPrices = nameof(ExistingPrices);
+
+        private readonly IPricingSearchService _pricingSearchService;
+        private readonly bool _not;
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: It is a very meaningless field name. Please, try to improve it.

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        public ImportProductPricesExistenceValidator(IPricingSearchService pricingSearchService, bool not = false)
+        {
+            _pricingSearchService = pricingSearchService;
+            _not = not;
+            AttachValidators();
+        }
+        public void AttachValidators()
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: private?

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs (hunk 1 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Fixed

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs (hunk 2 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Made internal

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs (hunk 3 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: It's enabled :) For some reason ReSharper doesn't apply these rules to newly generated class

Comment (asvishnyakov) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Fixed

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs (hunk 4 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Tried to fix

File: src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs (hunk 5 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Validation/ImportProductPricesExistanceValidator.cs]: Fixed

**Review by pushnitsa — APPROVED** [review]

**Review by trueboroda — APPROVED** [review]

**Comment by trueboroda** [comment]
These changes will be pushed with another PR.
https://github.com/VirtoCommerce/vc-module-simple-export-import/pull/25

**Comment by trueboroda** [comment]
These changes will be delivered with the next PR  https://github.com/VirtoCommerce/vc-module-simple-export-import/pull/25

**Pull request closed without merging**